### PR TITLE
feat(ui): add dark theme for editor and adjust some ui details

### PR DESCRIPTION
**Pull request by 100gle**
Now CodeMirror editor can automatically set dark theme in dark mode!

截屏2023-11-06 19 33 05

### CODE REVIEW

File: templates/challenge.html [hunk 1]
```diff
@@ -194,23 +200,31 @@
   </div>
 
   <script type="text/javascript">
+    let initTheme = localStorage.getItem('theme') === 'dark' ? "material-darker" : "default"
     let codeMirrorShared = {
```
Comment (laike9m) [inline templates/challenge.html:204]: `codeMirrorOptionsShared`

Comment (100gle) [reply templates/challenge.html:204]: Maybe we can rename this variable to `sharedCodeMirrorOptions` or `sharedCodeMirrorOpts`？🤔

Comment (laike9m) [reply templates/challenge.html:204]: `sharedCodeMirrorOptions` feels better

**Comment by laike9m** [comment]
Thanks a lot!

**Pull request merged**